docs/installation/install-binary.md

```diff
@@ -110,228 +106,703 @@ scp /usr/local/sbin/* karmada-02:/usr/local/sbin/
 
+## Step 1: Create Bash Scripts and Configuration Files
 
-## Generate certificate
+The following scripts will generate certificates using the `openssl` command. 
 
-Generated using the `openssl` command. Note yes `DNS` and `IP` when generating the certificate.
+We separate CA & leaf certificates generation script, so when you need to change Subject Alternative Name of leaf certificates (aka Load Balancer IP), you can reuse CA certificates, and run generate_leaf.sh to generate leaf certificates only.
 
-Execute operations at `karmada-01`.
 
-#### create a temporary directory for certificates
 
-```bash
-mkdir certs
-cd certs
+There are 3 CAs: front-proxy-ca, server-ca, etcd/ca. Why we need 3 CAs please see: [PKI certificates and requirements](https://kubernetes.io/docs/setup/best-practices/certificates/), [CA Reusage and Conflicts](https://kubernetes.io/docs/tasks/extend-kubernetes/configure-aggregation-layer/#ca-reusage-and-conflicts)
+
+If you use etcd provided by users, you can ignore `generate_etcd.sh` and `csr_config/etcd`.
+
+
+
+The file structure should look like this:
+
+```
+├── csr_config													# certificate signing request config file.
+│   ├── admin.conf											# kubectl => kube-apiserver
+│   ├── etcd														# certificate related to etcd
+│   │   ├── apiserver-etcd-client.conf	# kube-apiserver => etcd
+│   │   ├── healthcheck-client.conf			# etcdctl => etcd
+│   │   ├── peer.conf										# etcd peer => etcd peer
+│   │   └── server.conf									# etcd server
+│   ├── front-proxy-client.conf					# kube-apiserver => karmada-aggregated-apiserver
+│   ├── karmada.conf										# most karmada component use this
+│   ├── kube-apiserver.conf							# tls server certificate of kube-apiserver
+│   └── kube-controller-manager.conf		# tls client & server cert of kube-controller-manager
+├── generate_ca.sh											# genererate front-proxy-ca, server-ca
+├── generate_etcd.sh    								# genererate CA & leaf certificates of etcd.
+├── generate_leaf.sh    								# generate leaf certificates of front-proxy-ca, server-ca
+└── util.sh
 ```
 
-#### Create root certificate
 
-valid for 10 years
+
+(1) generate_ca.sh
 
 ```bash
-openssl genrsa -out ca.key 2048
-openssl req -x509 -new -nodes -key ca.key -subj "/C=CN/ST=Guangdong/L=Guangzhou/O=karmada/OU=System/CN=karmada" -days 3650 -out ca.crt
+#!/bin/bash
+
+set -e
+set -o pipefail
+
+function gen_server_ca() {
+	openssl genrsa -out server-ca.key 2048
+	openssl req -x509 -new -nodes -key server-ca.key -subj "/C=CN/ST=Guangdong/L=Guangzhou/O=karmada/OU=System/CN=karmada" -days 3650 -out server-ca.crt
+}
+
+function gen_front_proxy_ca() {
+	openssl genrsa -out front-proxy-ca.key 2048
+	openssl req -x509 -new -nodes -key front-proxy-ca.key -subj "/C=CN/ST=Guangdong/L=Guangzhou/O=karmada/OU=System/CN=front-proxy-ca" -days 3650 -out front-proxy-ca.crt
+}
+
+function main() {
+	mkdir ca_cert
+    cd ca_cert
+
+    gen_server_ca
+    gen_front_proxy_ca
+}
+
+main "$@"
```

Comment:
what's the function to give parameters from command lines?

Reply:
What do you mean? 

Actually, I use this style: <https://google.github.io/styleguide/shellguide.html#s7.8-main>

Reply:
You are right, keep this